components/src/core/Drawer/DrawerRailItem.tsx

```diff
@@ -0,0 +1,259 @@
+            left: 0,
+            width: 6,
+            zIndex: 100,
+            backgroundColor: (props): string => props.statusColor,
+        },
+        title: {
+            lineHeight: `16px`,
+            wordBreak: 'break-word',
+            hyphens: 'auto',
+            zIndex: 200,
+            color: (props): string =>
+                props.itemID !== props.activeItem
+                    ? props.itemFontColor || theme.palette.text.primary
+                    : props.activeItemFontColor ||
+                      (theme.palette.type === 'light' ? theme.palette.primary.main : lightenedPrimary),
+        },
+        titleActive: {
+            fontWeight: 600,
+        },
+        ripple: {
+            backgroundColor: theme.palette.primary.main,
+        },
+    });
+});
+
+const DrawerRailItemRender: React.ForwardRefRenderFunction<unknown, DrawerRailItemProps> = (
+    props: DrawerRailItemProps,
+    ref: any
+) => {
+    const {
+        activeItem,
+        classes = {},
+        condensed = false,
+        hidden,
+        icon,
+        itemID,
+        onClick,
+        onItemSelect,
+        ripple = true,
+        title,
+        /* eslint-disable @typescript-eslint/no-unused-vars */
+        activeItemBackgroundColor,
+        activeItemFontColor,
+        activeItemIconColor,
+        divider,
+        itemFontColor,
+        itemIconColor,
+        statusColor,
+        /* eslint-disable @typescript-eslint/no-unused-vars */
+
+        ...buttonBaseProps
+    } = props;
+
+    const defaultClasses = useStyles(props);
+
+    const active = activeItem === itemID;
+    const hasAction = Boolean(onItemSelect || onClick);
+
+    // Customize the color of the Touch Ripple
+    const RippleProps =
+        ripple && hasAction
+            ? {
+                  TouchRippleProps: {
+                      classes: {
+                          child: defaultClasses.ripple,
+                      },
+                  },
+              }
+            : {};
+
+    const combine = useCallback(
+        (className: keyof DrawerRailItemClasses): string => clsx(defaultClasses[className], classes[className]),
+        [defaultClasses, classes]
+    );
+
+    const getIcon = useCallback((): JSX.Element | undefined => {
+        if (icon) {
+            return <Avatar className={combine('icon')}>{icon}</Avatar>;
+        }
+    }, [icon, combine]);
+
+    const onClickAction = useCallback(
+        (e: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
+            if (onItemSelect) {
+                onItemSelect();
+            }
+            if (onClick) {
+                onClick(e);
+            }
+        },
+        [onItemSelect, onClick]
+    );
+
+    const innerContent = (
+        <ButtonBase
+            ref={ref}
+            {...buttonBaseProps}
+            className={clsx(defaultClasses.root, classes.root, {
+                [defaultClasses.condensed]: condensed,
+                [classes.condensed]: condensed,
```

Comment:
This appears to be a larger issue. Any time a class is optional (in the `classes` prop) and it is not provided, it's putting undefined. You can see this happening in the `activeTitle` class too on the title.

I can see about addressing this for the DrawerRailItem, but we should log another issue to fix this in the other components.

Reply:
it doesn't happen when you use clsx standard syntax (falseys are discarded), but when you use the object syntax to do things conditionally, we need an extra check to see if the class is defined (instead of undefined).

Reply:
so can we do `[classes.condensed]: condensed & classes?.condensed`?